### ISSUE TITLE
Subscriber does not correctly filter messages by the name of the service

### ISSUE THREAD

**Issue by seidlj** [pytroll/posttroll#24]
If I understand it correctly, the class Subscribe should return only messages produced by the services that were specified in the init parameter _services_. However, if I run a new publisher after the Subscriber was created, the subscriber starts listening to it regardless of its service name.

The first publisher:
```
from posttroll.publisher import Publish
from posttroll.message import Message

with Publish("service_1") as pub:
    while True:
        message = Message("/some_topic", "info", "service_1")
        pub.send(message.encode())
        time.sleep(3)
```
Subscriber:
```
from posttroll.subscriber import Subscribe
with Subscribe("service_1", "/some_topic", addr_listener=True) as sub:
    for msg in sub.recv(1):
        if msg is not None:
            print(msg)
```
It starts getting messages from 'service_1', topic '/some_topic'. However, if I run a second publisher 'service_2' publishing messages with the same topic,
```
from posttroll.publisher import Publish
from posttroll.message import Message

with Publish("service_2") as pub:
    while True:
        message = Message("/some_topic", "info", "service_2")
        pub.send(message.encode())
        time.sleep(3)
```
the subscriber returns them as well.

Debug info shown after creating 'service_2'
> DEBUG:posttroll.subscriber:Adding address tcp://my_ip:56679 service_1

displays a wrong service name. This indicates that this behaviour may be related to the following part of the code where the variable 'service' is overwritten by the for loop:

https://github.com/pytroll/posttroll/blob/516725eb0de548026d9597f7fb2fbda31ccebede/posttroll/subscriber.py#L405-L411

**Comment by mraspaud** [comment]
@seidlj Thank you for reporting this issue, and welcome! You have probably found a bug indeed...

Did you try renaming the loop variables to see if that fixed the problem ? But I suspect your diagnostic is correct. If that fixed it and you have the possibility, a PR would be very welcome :)

**Comment by seidlj** [comment]
Thank you for a quick response. I've verified that fixing the bug fixes the issue and submitted a PR #25 . The CI tests failed but it seems that they fail on the same line also in other PRs that are not related to the changes I've done.

**Comment by mraspaud** [comment]
Thanks so much for helping out, PR #25 is now merge and fixes this.
